src/NuGet.Clients/NuGet.PackageManagement.UI/ViewModels/SolutionPickerViewModel.cs

```diff
@@ -0,0 +1,109 @@
+            SolutionList = new ObservableCollection<SolutionPickerItemViewModel>();
+            OpenSolutionCommand = new SolutionOpenCommand(this);
+        }
+
+        public async Task PopulateSolutionListAsync(CodeContainerStorageManagerFactory storageManagerFactory)
+        {
+            if (storageManagerFactory == null)
+            {
+                throw new ArgumentNullException(nameof(storageManagerFactory));
+            }
+
+            ICodeContainerStorageManager manager = await storageManagerFactory.CreateAsync();
+
+            ActionBlock<StatefulReadOnlyList<CodeContainer, int?>> actionBlock = new ActionBlock<StatefulReadOnlyList<CodeContainer, int?>>(
+                action: async codeContainerList =>
+                {
+                    await NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+                    {
+                        var list = codeContainerList.List;
+                        await FilterListAsync(list);
+                    });
+                });
+            await manager.SubscribeAsync(actionBlock, CancellationToken.None);
+        }
+
+        private async Task FilterListAsync(IReadOnlyList<CodeContainer> list)
+        {
+            List<CodeContainer> solutionList = list.OrderByDescending(cc => cc.LastAccessed).Take(_maxResults).ToList();
+            await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+            foreach (CodeContainer cc in solutionList)
+            {
+                SolutionList.Add(new SolutionPickerItemViewModel(OpenSolutionCommand, cc.LocalProperties.FullPath));
+            }
+        }
+
+        private void RaiseSolutionClicked()
+        {
+            SolutionClicked?.Invoke(this, EventArgs.Empty);
+        }
+
+        public ObservableCollection<SolutionPickerItemViewModel> SolutionList { get; }
+
+        public ICommand OpenSolutionCommand { get; }
+
+        private class SolutionOpenCommand : ICommand
+        {
+            SolutionPickerViewModel _model;
+
+            public SolutionOpenCommand(SolutionPickerViewModel model)
+            {
+                _model = model;
+            }
+
+            // never raised
+            public event EventHandler CanExecuteChanged
+            {
+                add { CommandManager.RequerySuggested += value; }
+                remove { CommandManager.RequerySuggested -= value; }
+            }
+
+            public bool CanExecute(object parameter) => true;
+
+            public void Execute(object parameter)
+            {
+                ThreadHelper.ThrowIfNotOnUIThread();
+                var filePath = parameter.ToString();
+                if (File.Exists(filePath))
+                {
+                    var dte = Package.GetGlobalService(typeof(_DTE)) as DTE2;
+                    if (dte != null)
+                    {
+                        dte.Solution.Open(filePath);
+                        SolutionPickerViewModel model = new SolutionPickerViewModel();
```

Comment:
Why are you creating a new instance of the SolutionPickerViewModel? It looks like you're not using it anywhere.